### PR TITLE
Add /authenticate endpoint

### PR DESCRIPTION
**Pull request by karthick-vinod**

### CODE REVIEW

File: iam_openapi_spec.yml [hunk 1]
```diff
@@ -1045,6 +1045,33 @@ paths:
...
 
+  /authenticate:
+    post:
+      tags:
+        - User Authorization
+      summary: Authenticate a request
+      description: >
+        Authenticates the request and respond with token. Upon successful authentication,
+        - For basic auth as well as credential based bearer auth, this API generates a token and returns it.
+        - For JWT bearer auth, returns the same JWT token in response
```
Comment (hari-hypto) [inline iam_openapi_spec.yml:1056]: will this handle expiry check for JWT tokens ?

File: src/main/kotlin/com/hypto/iam/server/apis/TokenApi.kt [hunk 2]
```diff
@@ -48,4 +56,11 @@ fun Route.tokenApi() {
             generateToken(principal, call, context.request.accept())
         }
     }
+
+    authenticate("basic-auth", "unique-basic-auth", "bearer-auth") {
+        post("/authenticate") {
```
Comment (hari-hypto) [inline src/main/kotlin/com/hypto/iam/server/apis/TokenApi.kt:61]: Can we add a test for this.

**Review by hari-hypto — COMMENTED** [review]
Logic looks good to me.

File: iam_openapi_spec.yml (hunk 1 above)
Comment (karthick-vinod) [reply iam_openapi_spec.yml:1056]: Yes

File: src/main/kotlin/com/hypto/iam/server/apis/TokenApi.kt (hunk 2 above)
Comment (karthick-vinod) [reply src/main/kotlin/com/hypto/iam/server/apis/TokenApi.kt:61]: Copy of /getToken . So, skipping tests

**Pull request merged**